content/blog/2023-09-07-making-connecting-to-localstack-easier/index.md

```diff
@@ -0,0 +1,184 @@
+    * when using the gateway domain (`host.docker.internal`), this domain is also not included in the set of certificate domains.
+* Subdomains created by resources such as S3 buckets or OpenSearch clusters will not resolve to the LocalStack container.
+* Servers listening on the host must bind to either the Docker network gateway IP address, which is inconvenient to get, or `0.0.0.0` which exposes the server to traffic from outside the host, which may be insecure.
+* Each host port can only be bound once, whereas container ports are separate from each other and multiple containers can bind to the same port.
+
+We already resolve the first issue by using the domain name `localhost.localstack.cloud` in our documentation and examples.
+This domain name is publicly registered and resolves to the IP address `127.0.0.1`.
+This allows us to present a valid TLS certificate when using HTTPS from the host, but does not remove the connectivity problem.
+You can check that the domain maps to `127.0.0.1` by running:
+
+```sh
+dig @8.8.8.8 localhost.localstack.cloud
+```
+
+which outputs:
+
+```text
+; <<>> DiG 9.10.6 <<>> @8.8.8.8 localhost.localstack.cloud
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 54676
+;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 512
+;; QUESTION SECTION:
+;localhost.localstack.cloud.	IN	A
+
+;; ANSWER SECTION:
+localhost.localstack.cloud. 600	IN	A	127.0.0.1
+
+;; Query time: 54 msec
+;; SERVER: 8.8.8.8#53(8.8.8.8)
+;; WHEN: Fri Sep 08 11:23:20 BST 2023
+;; MSG SIZE  rcvd: 71
+```
+
+This command queries the Google public nameserver (`8.8.8.8`) for the `localhost.localstack.cloud` domain.
+In the "ANSWER" section we see `127.0.0.1` returned, as an `A` record, meaning IP address.
+
+### Why is using localhost.localstack.cloud not enough?
+
+When you create a lambda function, ECS container or EC2 instance, we create a new Docker container running your application code.
+In these situations, using the domain name `localhost.localstack.cloud` will not resolve to the LocalStack container as you may expect.
+We currently have code in our Lambda managed runtimes to intercept any domain names that resolve to `127.0.0.1` and rewrite them to the LocalStack container IP, but this is only available to managed runtimes.
+
+To tackle the problem generally, we are bringing a feature from LocalStack Pro into LocalStack open source: our DNS server.
+By doing this, we are able to respond with the IP address of the container for any requests to `localhost.localstack.cloud`, provided your code is running in a correctly configured environment.
+
+We are now able to resolve the three issues mentioned above:
+
+* If SSL is used, then certificate validation must be turned off since LocalStack does not present a valid certificate for the domain used(either `localhost` or `host.docker.internal`).
```

Comment:
```suggestion
* If SSL is used, then certificate validation must be turned off since LocalStack does not present a valid certificate for the domain used (either `localhost` or `host.docker.internal`).
```